src/internal/transaction_submission.ts

```diff
@@ -0,0 +1,163 @@
+/**
+ * This file contains the underlying implementations for exposed API surface in
+ * the {@link api/transaction_submission}. By moving the methods out into a separate file,
+ * other namespaces and processes can access these methods without depending on the entire
+ * transaction_submission namespace and without having a dependency cycle error.
+ */
+
+import { AptosConfig } from "../api/aptos_config";
+import { postAptosFullNode } from "../client";
+import { Account } from "../core/account";
+import { AccountAuthenticator } from "../transactions/authenticator/account";
+import {
+  buildTransaction,
+  generateTransactionPayload,
+  generateSignedTransactionForSimulation,
+  generateSignedTransaction,
+  sign,
+} from "../transactions/transaction_builder/transaction_builder";
+import { GenerateTransactionInput, AnyRawTransaction, SimulateTransactionData } from "../transactions/types";
+import { UserTransactionResponse, PendingTransactionResponse, MimeType } from "../types";
+
+/**
+ * Generates any transaction by passing in the required arguments
+ *
+ * @param args.sender The transaction sender's account address as a HexInput
+ * @param args.data EntryFunctionData | ScriptData | MultiSigData
+ * @param feePayerAddress optional. For a fee payer (aka sponsored) transaction
+ * @param secondarySignerAddresses optional. For a multi agent or fee payer (aka sponsored) transactions
+ * @param args.options optional. GenerateTransactionOptions type
+ *
+ * @example
+ * For a single signer entry function
+ * move function name, move function type arguments, move function arguments
+ * `
+ * data: {
+ *  function:"0x1::aptos_account::transfer",
+ *  type_arguments:[]
+ *  arguments:[recieverAddress,10]
+ * }
+ * `
+ *
+ * @example
+ * For a single signer script function
+ * module bytecode, move function type arguments, move function arguments
+ * ```
+ * data: {
+ *  bytecode:"0x001234567",
+ *  type_arguments:[],
+ *  arguments:[recieverAddress,10]
+ * }
+ * ```
+ *
+ * @return A raw transaction type (note that it holds the raw transaction as a bcs serialized data)
+ * ```
+ * {
+ *  rawTransaction: Uint8Array,
+ *  secondarySignerAddresses? : Array<AccountAddress>,
+ *  feePayerAddress?: AccountAddress
+ * }
+ * ```
+ */
+export async function generateTransaction(
+  args: { aptosConfig: AptosConfig } & GenerateTransactionInput,
+): Promise<AnyRawTransaction> {
+  const { aptosConfig, sender, data, options, secondarySignerAddresses, feePayerAddress } = args;
+  const payload = await generateTransactionPayload(data);
+  const rawTransaction = await buildTransaction({
```

Comment:
This is different than buildTransaction, or should it also be named buildTransaction?